internal/shared/validations/validations.go

```diff
@@ -0,0 +1,71 @@
+package validations
+
+import (
+	"context"
+	"net/url"
+	"regexp"
+)
+
+// Valid git transports url
+
+var (
+	Transports = NewTransportSet(
+		"ssh",
+		"git",
+		"git+ssh",
+		"http",
+		"https",
+		"ftp",
+		"ftps",
+		"rsync",
+		"file",
+	)
+)
+
+// Checks for valid remote repository
+
+type TransportSet struct {
+	Transports map[string]struct{}
+}
+
+// NewTransportSet returns a TransportSet with the items keys mapped
+// to empty struct values.
+func NewTransportSet(items ...string) *TransportSet {
+	t := &TransportSet{
+		Transports: map[string]struct{}{},
+	}
+	for _, i := range items {
+		t.Transports[i] = struct{}{}
+	}
+	return t
+}
+
+// Valid returns true if transport is a known Git URL scheme and false
+// if not.
+func (t *TransportSet) Valid(transport string) bool {
+	_, ok := t.Transports[transport]
+	return ok
+}
+
+func CheckGitURL(gitURL string, ctx context.Context) bool {
+	u, err := url.Parse(gitURL)
+	if err == nil && !Transports.Valid(u.Scheme) {
+		return false
+	}
+	return true
+}
+
+func IsValidKubernetesNamespace(name string) bool {
+	if len(name) > 63 {
+		return false
+	}
+	// names can only contain lowercase alphanumeric characters, '-', but must start and end with an alphanumeric
+	validNameRegex := regexp.MustCompile(`^[a-z0-9][a-z0-9\-]*[a-z0-9]$`)
+	return validNameRegex.MatchString(name)
+}
+
+func IsReservedName(name string) bool {
```

Comment:
Can you also find is there any built in support to check Reserved name as well? Thanks!